src/Process/Runner/ParallelProcessRunner.php

```diff
@@ -92,6 +99,10 @@ public function run(iterable $processes): void
 
         // start the initial batch of processes
         while ($process = array_shift($bucket)) {
+            if ($this->shouldStop) {
```

Comment:
This is exactly how Symfony Messenger handles the same signals. Basically, the idea and implementation are similar to Symfony's one.

See 

- https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Messenger/EventListener/DispatchPcntlSignalListener.php
- https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Messenger/EventListener/StopWorkerOnSignalsListener.php
- https://github.com/symfony/symfony/blob/8c637a5bbfea8b322ec93eb51ac47c0db8948e2a/src/Symfony/Component/Messenger/Worker.php#L111